lib/ts/recipe/thirdparty/components/themes/signInAndUp/providersForm.tsx

```diff
@@ -64,9 +66,44 @@ export const ThirdPartySignInAndUpProvidersForm: React.FC<SignInAndUpThemeProps>
         }
     };
 
+    const providers = useMemo(() => {
+        const usesDynamicLoginMethods = SuperTokens.usesDynamicLoginMethods === true;
+        if (usesDynamicLoginMethods === false) {
+            return props.providers.map((provider) => ({
+                id: provider.id,
+                buttonComponent: provider.getButton(),
+            }));
+        }
+        const tenantProviders = ThirdParty.tenantProviders || [];
+        const providers: { id: string; buttonComponent: JSX.Element }[] = [];
+        for (const tenantProvider in tenantProviders) {
+            let provider = props.providers.find((provider) => {
+                const { id } = tenantProviders[tenantProvider];
+                return provider.id === id || provider.id.includes(id);
+            });
+            if (provider === undefined) {
+                provider = props.providers.find((provider) => {
+                    const { id } = tenantProviders[tenantProvider];
+                    return id.includes(provider.id);
+                });
+            }
+            if (provider !== undefined) {
+                if (provider.id !== tenantProviders[tenantProvider].id) {
+                    provider.setId(tenantProviders[tenantProvider].id);
+                }
+                provider.setName(tenantProviders[tenantProvider].name);
```

Comment:
Setting the id and name possibly multiple times doesn't feel right. Did you test this with multiples of the same provider type added with different names?